metadata-ingestion/src/datahub/ingestion/run/classifier_pipeline.py

```diff
@@ -0,0 +1,252 @@
+        if self.config.transformers is not None:
+            for transformer in self.config.transformers:
+                transformer_type = transformer.type
+                transformer_class = transform_registry.get(transformer_type)
+                transformer_config = transformer.dict().get("config", {})
+                self.transformers.append(
+                    transformer_class.create(transformer_config, self.ctx)
+                )
+                logger.debug(
+                    f"Transformer type:{transformer_type},{transformer_class} configured"
+                )
+
+    def _get_schema_name(self, record: RecordEnvelope) -> str:
+        res = list(filter(lambda x: isinstance(x, SchemaMetadataClass), record.record.proposedSnapshot.aspects))
+        return res[0].schemaName
+
+    def _consistent_hash(self, s: str) -> int:
+        # these hashes will be the same across runs / processes, as opposed to python's built in `hash` function
+        return abs(int(hashlib.sha256(s.encode('utf-8')).hexdigest(), 16) % 10**8)
+
+    def _get_dataset_pii_classification_state_record(self, urn: str):
+        response = self.pii_classification_state_table.query(
+            KeyConditionExpression=boto3.dynamodb.conditions.Key('pk').eq(urn) & boto3.dynamodb.conditions.Key('sk').eq(self.DATASET_RECORD_SK)
+        )
+        return None if len(response['Items']) == 0 else response['Items'][0]
+
+    def _write_classification_result(self, urn: str, classification_result: Dict[str, ClassificationResult], is_first_time_classified: bool):
+        now = datetime.datetime.now()
+
+        are_writes_successful = True
+        for col_name, class_result in classification_result.items():
+            for law in class_result.privacy_laws:
+                are_writes_successful &= self._attach_glossary_terms(urn, col_name, "PrivacyLaw.{}".format(law))
+            for type in class_result.pii_types:
+                are_writes_successful &= self._attach_glossary_terms(urn, col_name, "PiiData.{}".format(type))
+
+        if are_writes_successful:
+            if is_first_time_classified:
+                self.pii_classification_state_table.put_item(
+                    Item={
+                            'pk': urn,
+                            'sk': self.DATASET_RECORD_SK,
+                            'createdDate': now.isoformat(),
+                            'lastEvalDate': now.isoformat()
+                        }
+                    )
+            else:
+                self.pii_classification_state_table.update_item(
+                    Key={
+                        'pk': urn,
+                        'sk': self.DATASET_RECORD_SK
+                    },
+                    UpdateExpression="SET lastEvalDate = :newEvalDate",
+                    ExpressionAttributeValues={":newEvalDate": now.isoformat()}
+                )
+
+    def _attach_glossary_terms(self, urn: str, subfield: str, glossary_term: str) -> bool:
```

Comment:
One thing that could be done differently is doing additional work to support the same `sink` configurations (and allow us to attach glossary terms via kafka, GMS, etc, as opposed to just Frontend as it is here). This might be purer and more in-line with how ingestion is implemented, but would require additional work to pull existing dataset snapshots to see existing / custom glossary terms attached to each dataset (without this extra step, we would overwrite existing attached glossary terms with the result of each classifier run)

I don't see the extra work being worth it, but open to discussing further (publishing via frontend API is simple and gets the job done).